core/common-util/src/main/java/uk/gov/gchq/gaffer/commonutil/iterable/EmptyIterable.java

```diff
@@ -19,23 +19,26 @@
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 
+import java.io.Closeable;
+import java.util.Iterator;
+
 /**
- * An {@code EmptyCloseableIterable} is an {@link java.lang.Iterable} which is backed by a
- * {@link EmptyCloseableIterator}, and contains no objects.
+ * An {@code EmptyIterable} is an {@link java.io.Closeable} {@link java.lang.Iterable}
+ * which is backed by a {@link EmptyIterator}, and contains no objects.
  *
  * This is useful when a {@link java.lang.Iterable} is required, but there is no data present.
  *
  * @param <T> the type of items in the iterable.
  */
-public class EmptyClosableIterable<T> implements CloseableIterable<T> {
+public class EmptyIterable<T> implements Closeable, Iterable<T> {
```

Comment:
Possibly... As you can't return both types and `Iterable` is preferred to `Closeable` I changed it

Reply:
Looking at the `Iterable` `SuppliedIterable`, it implemented `CloseableIterable` but was still kept it's `Iterable` name. Which precedence do we want to follow?

Reply:
You're right it seems most don't have this, just wondering if it might make sense to keep the ones that were called Closeable before as Closeable. Will make migration easier but might also be worth adding breaking changes now?

Reply:
After a discussion, we think that renaming the classes here to keep them consistent is a good place to start, as it's an alpha.

Reply:
Worse case, if they need to be changed back for compatibility, they can be in a later alpha.